core/google/api/core/gapic_v1/config.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Helpers for loading gapic configuration data.
+
+The Google API generator creates supplementary configuration for each RPC
+method to tell the client library how to deal with retries and timeouts.
+"""
+
+import collections
+
+import grpc
+import six
+
+from google.api.core import exceptions
+
+
+_MILLIS_PER_SECOND = 1000.0
+
+
+MethodConfig = collections.namedtuple('MethodConfig', [
+    'timeout',
+])
+
+RetryableMethodConfig = collections.namedtuple('RetryableMethodConfig', [
+    # Retry settings
+    'retry_exceptions',
+    'initial_delay',
+    'delay_multiplier',
+    'max_delay',
+    'deadline',
+    # Timeout settings
+    'initial_timeout',
+    'timeout_multiplier',
+    'max_timeout',
+])
+
+
+def _exception_class_for_grpc_status_name(name):
+    """Returns the Google API exception class for a gRPC error code name."""
+    return exceptions.exception_class_for_grpc_status(
+        getattr(grpc.StatusCode, name))
+
+
+def _make_retryable_config(retry_params, retry_codes):
+    """Creates a retryable method configuration.
+
+    Args:
+        retry_params (dict): The retry parameter values, for example::
+
+            {
+                "initial_retry_delay_millis": 1000,
+                "retry_delay_multiplier": 2.5,
+                "max_retry_delay_millis": 120000,
+                "initial_rpc_timeout_millis": 120000,
+                "rpc_timeout_multiplier": 1.0,
+                "max_rpc_timeout_millis": 120000,
+                "total_timeout_millis": 600000
+            }
+
+        retry_codes (sequence[str]): The list of retryable gRPC error code
+            names.
+
+    Returns:
+        RetryableMethodConfig: Configuration for the method.
```

Comment:
Should you have a `Raises` section saying `KeyError` will occur unless the caller sends all the keys mentioned above?